Two step Feedback flow (#3154)
mapbox/mapbox-navigation-ios: CHANGELOG.md

* Fixed a retain cycle in `UserCourseView`. ([#3120](https://github.com/mapbox/mapbox-navigation-ios/issues/3120))
* The `EventsManagerDataSource.router`, `NavigationService.router`, `NavigationService.eventsManager`, `MapboxNavigationService.router`, `MapboxNavigationService.eventsManager` properties are no longer force unwrapped. ([#3055](https://github.com/mapbox/mapbox-navigation-ios/pull/3055))
* Fixed an issue where traffic congestion segments along the route line blurred into each other when the map was zoomed in far enough. ([#3153](https://github.com/mapbox/mapbox-navigation-ios/pull/3153))
* Supported feedbacks during passive navigation. Set `PassiveLocationManager` as a data source for `NavigationEventsManager` and then use this class to create and send user feedback. `FeedbackViewController` provides a drop-in feedback screen and can be used during both passive and active navigation. ([#3122](https://github.com/mapbox/mapbox-navigation-ios/pull/3122))

> * Added `NavigationEventsManager.createFeedback()` method that now returns an instance of `FeedbackEvent`. Application can update this model with additional information and send to the Mapbox data team using new `NavigationEventsManager.sendFeedback(_:type:description:)` method. Use this API if you have custom feedback UI. If you use drop-in UI from `FeedbackViewController` feedback lifecycle will be managed internally. ([#3154](https://github.com/mapbox/mapbox-navigation-ios/pull/3154))

Author: Thanks, updated the changelog according to your proposal, just added a new item about deferred feedback. Complete changelog now:

```
* You can now solicit user feedback about `PassiveLocationManager` and `NavigationMapView` outside of active turn-by-turn navigation. Set `PassiveLocationManager` as a data source for a `NavigationEventsManager`, then use `NavigationEventsManager` to create and send user feedback. You can use a `FeedbackViewController` to present the user with the same options as during turn-by-turn navigation. Alternatively, if you present a custom feedback UI, call the `NavigationEventsManager.createFeedback()` method and configure the resulting `FeedbackEvent` with any additional context. ([#3122](https://github.com/mapbox/mapbox-navigation-ios/pull/3122))
* You can now manage the feedback event lifecycle allowing users to submit additional details later. Use `NavigationEventsManager.createFeedback()` to create a `FeedbackEvent` and `NavigationEventsManager.sendFeedback(_:type:description:)` to send it to the Mapbox. `FeedbackEvent` conforms to the `Codable` protocol, so the application can store uncompleted feedbacks persistently. [#3154](https://github.com/mapbox/mapbox-navigation-ios/pull/3154)
```

* The `EventsManagerDataSource.router`, `NavigationService.router`, `NavigationService.eventsManager`, `MapboxNavigationService.router`, `MapboxNavigationService.eventsManager` properties are no longer force unwrapped. ([#3055](https://github.com/mapbox/mapbox-navigation-ios/pull/3055))
* Fixed an issue where traffic congestion segments along the route line blurred into each other when the map was zoomed in far enough. ([#3153](https://github.com/mapbox/mapbox-navigation-ios/pull/3153))
* You can now solicit user feedback about `PassiveLocationManager` and `NavigationMapView` outside of active turn-by-turn navigation. Set `PassiveLocationManager` as a data source for a `NavigationEventsManager`, then use `NavigationEventsManager` to create and send user feedback. You can use a `FeedbackViewController` to present the user with the same options as during turn-by-turn navigation. Alternatively, if you present a custom feedback UI, call the `NavigationEventsManager.createFeedback()` method and configure the resulting `FeedbackEvent` with any additional context. ([#3122](https://github.com/mapbox/mapbox-navigation-ios/pull/3122))

> * You can now manage the feedback event lifecycle allowing users to submit additional details later. Use `NavigationEventsManager.createFeedback()` to create a `FeedbackEvent` and `NavigationEventsManager.sendFeedback(_:type:description:)` to send it to the Mapbox. `FeedbackEvent` conforms to the `Codable` protocol, so the application can store uncompleted feedbacks persistently. [#3154](https://github.com/mapbox/mapbox-navigation-ios/pull/3154)

Reviewer: I like the mental picture of putting feedback in the MapBox. 🙃

```suggestion
* You can now manage the feedback event lifecycle, allowing the user to submit additional details later. Use `NavigationEventsManager.createFeedback()` to create a `FeedbackEvent` and `NavigationEventsManager.sendFeedback(_:type:description:)` to send it to Mapbox. `FeedbackEvent` conforms to the `Codable` protocol, so your application can store incomplete feedback across sessions if necessary. ([#3154](https://github.com/mapbox/mapbox-navigation-ios/pull/3154))
```
